src/features/HallLayout/components/BuyTicketModal/BuyTicketModal.tsx

```diff
@@ -0,0 +1,235 @@
+import { FoodAndBeverage } from "../../../FoodAndBeverages/interfaces/FoodAndBeverage";
+import { useState } from "react";
+import { Dropdown, DropdownButton, Form, InputGroup } from "react-bootstrap";
+import { BuyTicketModalCheckoutStage } from "./BuyTicketModalCheckoutStage";
+import { useBuyTicket } from "../../hooks/useBuyTicket";
+Form;
+
+export interface SelectedItem {
+    item: FoodAndBeverage;
+    quantity: number;
+}
+
+interface Props {
+    showBuyTicketModal: boolean;
+    setShowBuyTicketModal: React.Dispatch<React.SetStateAction<boolean>>;
+    projection: Projection;
+    reservation?: Reservation;
+    selectedSeat: SelectedSeat | null;
+    setSelectedSeat: React.Dispatch<React.SetStateAction<SelectedSeat | null>>;
+    setProjection: React.Dispatch<React.SetStateAction<Projection>>;
+}
+
+export const BuyTicketModal: React.FC<Props> = ({
+    showBuyTicketModal,
+    projection,
+    reservation,
+    selectedSeat,
+    setShowBuyTicketModal,
+    setSelectedSeat,
+    setProjection,
+}) => {
+    const { foodAndBeverages } = useFoodAndBeverages();
+
+    const [modalStage, setModalStage] = useState(0);
+    const [selectedItems, setSelectedItems] = useState<{ [key: string]: SelectedItem }>({});
+    const [price, setPrice] = useState(0);
+
+    const handleSelect = (item: FoodAndBeverage) => {
+        setSelectedItems((prevSelectedItems) => ({
+            ...prevSelectedItems,
+            [item._id]: {
+                item,
+                quantity: prevSelectedItems[item._id] ? prevSelectedItems[item._id].quantity : 1,
+            },
+        }));
+    };
+
+    const handleIncreaseQuantity = (itemId: string) => {
+        setSelectedItems((prevSelectedItems) => ({
+            ...prevSelectedItems,
+            [itemId]: {
+                ...prevSelectedItems[itemId],
+                quantity: prevSelectedItems[itemId].quantity + 1,
+            },
+        }));
+    };
+
+    const handleDecreaseQuantity = (itemId: string) => {
+        setSelectedItems((prevSelectedItems) => ({
+            ...prevSelectedItems,
+            [itemId]: {
+                ...prevSelectedItems[itemId],
+                quantity: Math.max(1, prevSelectedItems[itemId].quantity - 1),
+            },
+        }));
+    };
+
+    const handleRemoveItem = (itemId: string) => {
+        setSelectedItems((prevSelectedItems) => {
+            const updatedItems = { ...prevSelectedItems };
+            delete updatedItems[itemId];
+            return updatedItems;
+        });
+    };
+
+    const { buyTicketHandler } = useBuyTicket(setProjection);
+
+    return (
+        <Modal show={showBuyTicketModal} size="lg" aria-labelledby="contained-modal-title-vcenter" centered>
+            <Modal.Header closeButton>
+                <Modal.Title id="contained-modal-title-vcenter">Modal heading</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+                {modalStage === 0 && (
```

Comment:
create an enum for modal stages and name each stage properly top make it more clear